src/babel-serializer/geometry/off.hh

```diff
@@ -1,3 +1,43 @@
 #pragma once
 
-// TODO
+#include <cstddef> // std::byte
+
+#include <clean-core/span.hh>
+#include <clean-core/typedefs.hh>
+#include <clean-core/vector.hh>
+
+#include <typed-geometry/tg-lean.hh>
+
+#include <babel-serializer/errors.hh>
+
+namespace babel::off
+{
+struct read_config
+{
+    bool has_colors = false;
+    bool has_normals = false;
+    bool has_tex_coords = false;
+    int dimension = 3;
+};
+struct geometry
+{
+    struct face_entry
+    {
+        int vertex_index = -1;
+    };
+    struct face
+    {
+        int entries_start = -1;
+        int entries_count = -1;
+    };
+
+    cc::vector<tg::pos4> vertices;
+    cc::vector<tg::vec3> normals;
+    cc::vector<tg::pos2> tex_coords;
+    cc::vector<face_entry> face_entries;
+    cc::vector<face> faces;
+    cc::vector<tg::color4> colors;
+    cc::vector<int> colormap_indices;
```

Comment:
how is this interpreted? what about a `cc::optional<tg::color3>` per `face`?

Reply:
Color information is extremely weird. The colormap is an optional additional file, that would require additional parsing. 
Generally regarding the colors, it might even be allowed to mix different color formats :/

Reply:
ok after reading `http://www.holmes3d.net/graphics/offfiles/offdesc.txt` I would say `optional<color4>` and we resolve the colormap in the parse?

Reply:
ok colormap is stretchgoal: that requires a separate file Oo

Reply:
Do we want to offer a default color?